src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3039 @@
+                self.apu_bleed_air_valve.is_open(),
+                overhead_panel,
+                engine_fire_push_buttons,
+                self.cross_bleed_valve.is_open(),
+                &self.fadec,
+            );
+
+            // I am not exactly sure if both BMCs should actually control this valve all the time.
+            self.cross_bleed_valve
+                .update_open_amount(&bleed_monitoring_computer.main_channel);
+        }
+
+        let (bleed_monitoring_computer_one, bleed_monitoring_computer_two) =
+            self.bleed_monitoring_computers.split_at_mut(1);
+
+        bleed_monitoring_computer_one[0].check_for_failure(&mut bleed_monitoring_computer_two[0]);
+        bleed_monitoring_computer_two[0].check_for_failure(&mut bleed_monitoring_computer_one[0]);
+
+        for engine_starter_valve_controller in self.engine_starter_valve_controllers.iter_mut() {
+            engine_starter_valve_controller.update(&self.fadec);
+        }
+
+        for engine_system in self.engine_systems.iter_mut() {
+            for bleed_monitoring_computer in self.bleed_monitoring_computers.iter() {
+                let index = engine_system.number - 1;
+
+                // If we get an actual channel here, this means that the channel is not in slave mode
+                if let Some(channel) =
+                    bleed_monitoring_computer.channel_for_engine(engine_system.number)
+                {
+                    engine_system.update(
+                        context,
+                        channel,
+                        channel,
+                        &self.engine_starter_valve_controllers[index],
+                        channel,
+                        engines[index],
+                    );
+                }
+            }
+        }
+
+        self.update_hydraulic_reservoir_spatial_volumes(
+            hydraulics.green_reservoir_capacity() - hydraulics.green_reservoir_volume(),
+            hydraulics.blue_reservoir_capacity() - hydraulics.blue_reservoir_volume(),
+            hydraulics.yellow_reservoir_capacity() - hydraulics.yellow_reservoir_volume(),
+        );
+
+        let (left_system, right_system) = self.engine_systems.split_at_mut(1);
+        self.apu_bleed_air_valve
+            .update_move_fluid(context, &mut self.apu, &mut left_system[0]);
+
+        self.cross_bleed_valve.update_move_fluid(
+            context,
+            &mut left_system[0],
+            &mut right_system[0],
+        );
+
+        self.green_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+        self.blue_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+        self.yellow_hydraulic_reservoir_with_valve
+            .update_flow_through_valve(context, &mut left_system[0]);
+
+        self.packs[0].update(context, &mut left_system[0]);
+        self.packs[1].update(context, &mut right_system[0]);
```

Comment:
**suggestion**: The code below is longer, but can be used to generalise to any number of packs and engines.
```suggestion
        self.packs
            .iter_mut()
            .zip(self.engine_systems.iter_mut())
            .for_each(|(pack, engine_system)| {
                pack.update(context, engine_system);
            });
```